### PR TITLE
Add openshift/build-machinery-go

### PR DESCRIPTION
**Pull request by tnozicka**
/cc @stevekuznetsov @droslean @petr-muller 

@openshift/sig-master

### CODE REVIEW

**Comment by tnozicka** [comment]
rehearsals are chicken-egg

File: ci-operator/config/openshift/build-machinery-go/openshift-build-machinery-go-master.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      memory: 3Gi
+tag_specification:
+  name: "4.4"
+  namespace: ocp
+tests:
+- as: verify
+  commands: |
+    make verify -k --warn-undefined-variables
+  container:
+    from: bin
```
Comment (droslean) [inline ci-operator/config/openshift/build-machinery-go/openshift-build-machinery-go-master.yaml]: Should be `from: src` . Thats why the rehearsal failed.

Comment (petr-muller) [reply ci-operator/config/openshift/build-machinery-go/openshift-build-machinery-go-master.yaml]: yeah, to have `bin` you need `binary_build_commands`.

File: core-services/prow/02_config/_config.yaml [hunk 2]
```diff
@@ -1105,6 +1105,7 @@ tide:
     - do-not-merge/invalid-owners-file
     - bugzilla/invalid-bug
     repos:
+    - openshift/build-machinery-go
```
Comment (petr-muller) [inline core-services/prow/02_config/_config.yaml]: Should this `openshift` repo really be outside of the BZ machinery?

File: ci-operator/config/openshift/build-machinery-go/openshift-build-machinery-go-master.yaml (hunk 1 above)
Comment (tnozicka) [reply ci-operator/config/openshift/build-machinery-go/openshift-build-machinery-go-master.yaml]: fixed

File: core-services/prow/02_config/_config.yaml (hunk 2 above)
Comment (tnozicka) [reply core-services/prow/02_config/_config.yaml]: yes, library-go has exception, the split inherits it.

Comment (tnozicka) [reply core-services/prow/02_config/_config.yaml]: see https://github.com/openshift/release/pull/7013

**Pull request merged**